plugins/azure-sites-common/src/permissions.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createPermission } from '@backstage/plugin-permission-common';
+
+/**
+ * @public
+ */
+export const azureSitesReadPermission = createPermission({
+  name: 'azure.sites.read',
+  attributes: { action: 'read' },
+});
+
+/**
+ * @public
+ */
+export const azureSitesActionPermission = createPermission({
+  name: 'azure.sites.update',
+  attributes: { action: 'update' },
```

Comment:
The `RESOURCE_TYPE_CATALOG_ENTITY` gives more conditional decisions like isEntityOwner/hasLabels. Example `isEntityOwner` can be a team but the start & stop action needs to be accessed by specific people/group with in team to avoid potential start&stop. With this Basic Permission lets the User decide their own approach to ALLOW the permission. Moreover The `router.post('/:subscription/:resourceGroup/:name/stop',.. ` don't gives the entity reference to pass it to rules and the `name` in request.param don't need to be same as entity name.

Reply:
But what if someone else in the community would like to perform more complex conditional decisions?
And actually, the code doesn't match what you are saying here. For example, since the permission is not a resource permission, you can't apply any conditional decision, meaning that you cannot even match subusers of a team owning the entity. You can only match specific users which are detached from owning a specific entity, meaning that a user can be allowed to start an stop ANY site but not a specific one.

Instead, you can achieve exactly what you're saying (plus much more) by using `RESOURCE_TYPE_CATALOG_ENTITY`: it's the permission policy who decides if an entire team should be allowed to execute the action, or if only a bunch of people should be allowed.

Reply:
I agree!
I will switch permission from Basic to Resource and use `RESOURCE_TYPE_CATALOG_ENTITY`.